lib/mongo/collection.ex

```diff
@@ -580,17 +580,23 @@ defmodule Mongo.Collection do
               unquote(attribute_names),
               %__MODULE__{},
               fn name, result ->
-                Map.put(result, name, map[Atom.to_string(name)])
+                Map.put(result, name, map[Atom.to_string(@attributes[name][:name])])
```

Comment:
Maybe we can put the mapping work out of the function like this:

```elixir
   load_function =
      quote unquote: false do
        attribute_names = Enum.map(@attributes, fn {name, opts} -> {name, Atom.to_string(opts[:name])} end)

...
        def load(map, false) when is_map(map) do
          struct = Enum.reduce(unquote(attribute_names), %__MODULE__{}, fn {name, src_name}, result -> Map.put(result, name, map[src_name]) end)
...
```

We create a list of tuples {name, src_name}, and then we can use the src_name directly in the reduce function.